### PR TITLE
test: swap + send eth e2e tests

### PR DESCRIPTION
**Pull request by micaelae**
## **Description**

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/24449?quickstart=1)

## **Related issues**

Fixes:

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: ui/components/multichain/pages/send/components/quote-card/hooks/useEthFeeData.tsx
Comment (BZahory) [inline ui/components/multichain/pages/send/components/quote-card/hooks/useEthFeeData.tsx:59]: Can we use `addHexPrefix` here instead

Comment (BZahory) [reply ui/components/multichain/pages/send/components/quote-card/hooks/useEthFeeData.tsx:59]: Side note but is this not a numeric value? We would need to change the base in the next parameter as well if it's a hex

**Review by georgewrmarshall — APPROVED** [review]
Not sure why this is blocked by @MetaMask/design-system-engineers, did it previously touch the component-library? Approving to unblock

File: ui/components/multichain/pages/send/components/quote-card/hooks/useEthFeeData.tsx
Comment (micaelae) [reply ui/components/multichain/pages/send/components/quote-card/hooks/useEthFeeData.tsx:59]: If you search for `GAS_PRICE_API_MOCK_RESULT` or `gasPricesCallbackMock` it looks like the prices are in decimals

The infura https://gas.api.infura.io/networks/1/gasPrices api might be deprecated tho bc I can't find docs on it

**Review by BZahory — APPROVED** [review]

**Pull request merged**
